### DISCUSSION TITLE
Errors with the inverse problem applied to some light curves

### DISCUSSION THREAD

**Discussion by JGLetelier** [phoebe-project/phoebe2#614] [Questions]
Hello,
I'm trying to model the light curves of some binary systems using the latest version of PHOEBE (v. 2.4.3). For this task I have followed the [Inverse Problem tutorial,](http://phoebe-project.org/docs/2.4/examples/inverse_paper_examples) which has been useful for some light curves. However, for other systems I have had different types of problems. For example,

Problem 1: for a given system I get the following error:

>>> b.run_solver(solver='lcgeom_solver', solution='lc_geom_sol'). 
2G model C failed with error: Residuals are not finite in the initial point.
2G model CE failed with error: Residuals are not finite in the initial point.
2G model CG failed with error: Residuals are not finite in the initial point.
2G model CGE failed with error: Residuals are not finite in the initial point.
2G model CG12 failed with error: Residuals are not finite in the initial point.
2G model CG12E1 failed with error: Residuals are not finite in the initial point.
2G model CG12E2 failed with error: Residuals are not finite in the initial point.
…

Problem 2: For another binary system I have the following error:

>>> b.adopt_solution('ebai_sol', adopt_parameters=['teffratio', 'requivsumfrac', 'incl'])
File "/home/user/.local/lib/python3.8/site-packages/phoebe/frontend/bundle.py", line 13799, in adopt_solution
    self.run_checks_solution(solution=solution_ps.solution,
File "/home/user/.local/lib/python3.8/site-packages/phoebe/frontend/bundle.py", line 5056, in run_checks_solution
    self._run_checks_warning_error(report, raise_logger_warning, raise_error)
File "/home/user/.local/lib/python3.8/site-packages/phoebe/frontend/bundle.py", line 3213, in _run_checks_warning_error
    raise ValueError("failed to pass checks\n{}".format(report))
ValueError: failed to pass checks
Run Checks Report: FAIL
ERROR: at least one parameter in adopt_parameters includes nan in fitted_values (2 affected parameters, affecting adopt_solution)

I suspect that for case 2 the problem goes back to the following warning:

>>>b.run_solver(kind='ebai', solution='ebai_sol')
/home/user/.local/lib/python3.8/site-packages/phoebe/solverbackends/solverbackends.py:1301: RuntimeWarning: invalid value encountered in arcsin
 fitted_values = [t0_supconj, teffratio, requivsumfrac, esinw, ecosw, np.arcsin(sini)]

I must emphasize that I have applied the same tutorial to other light curves without any problem. I would appreciate an answer to help me solve both problems please, since I need to model these systems using PHOEBE. Thanks!

Regards,
JGL.

**Comment by gecheline** [comment]
Hi JGL,

The first problem occurs due to the phase binning which leaves some empty bins that the two-Gaussian model can’t fit yet (I’m working on fixing this). For now you can resolve it by setting the value of the phase_nbins parameter to a lower value. The default is 500 so I suggest trying 200 or less. As of 2.4 you can also change the analytical model to polyfit by setting the analytical_model parameter to ‘polyfit’. This model doesn’t have the binning issue.

As for the second error, the only fix would be to manually override the adopt_parameters value to only the set of twigs that aren’t nans in the solution. You can check which those are by printing out the fitted_twigs and fitted_values for that solution. In 2.4 there is also an option to change the ebai_method to “knn” instead of the default “mlp” which may yield a better estimate without nans. 

Let me know if these suggestions work for you!

**Reply by JGLetelier** [reply]
Hello,

  I really appreciate your quick answer. The two tips were very helpful in solving the problem. I solved problem 1 using analytical_model='polyfit'. Problem 2 was solved using ebai_method='knn'.
  Thank you very much for your help.

Regards,
JGL.
